src/vmspawn/vmspawn.c

```diff
@@ -1002,9 +1040,20 @@ static int run(int argc, char *argv[]) {
         if (r < 0)
                 goto finish;
 
+        r = sd_listen_fds_with_names(true, &names);
+        if (r < 0)
+                goto finish;
+
+        for (size_t i = 0; i < strv_length(names); i++) {
```

Comment:
the return value of sd_listen_fds_with_names() already tells you the length of the array

Reply:
Fixed in 56f995436f.